src/core/utils/color/ColorValue.ts

```diff
@@ -0,0 +1,709 @@
+    // Check for simple case first.
+    if (!hsv.s || hsv.h === -1) {
+      // hue must be undefined, have no color only white
+      const white = 0xff & Math.floor((255.0 * hsv.v) / 100.0 + 0.5 + 3.0e-14);
+      return ColorValue.from(white, white, white, 0);
+    }
+
+    let dhue = hsv.h,
+      dsaturation = hsv.s,
+      dvalue = hsv.v;
+    if (dhue === 360) {
+      dhue = 0.0;
+    }
+
+    dhue /= 60; // hue is now [0..6]
+    const hueIntpart = Math.floor(dhue); // convert double -> int
+    const hueFractpart = dhue - hueIntpart;
+    dvalue /= 100;
+    dsaturation /= 100;
+
+    const p = 0xff & Math.floor(dvalue * (1.0 - dsaturation) * 255.0 + 0.5);
+    const q =
+      0xff &
+      Math.floor(dvalue * (1.0 - dsaturation * hueFractpart) * 255.0 + 0.5);
+    const t =
+      0xff &
+      Math.floor(
+        dvalue * (1.0 - dsaturation * (1.0 - hueFractpart)) * 255.0 + 0.5,
+      );
+    const v = 0xff & Math.floor(dvalue * 255 + 0.5);
+
+    let r = 0,
+      b = 0,
+      g = 0;
+    switch (hueIntpart) {
+      case 0:
+        r = v;
+        g = t;
+        b = p;
+        break; // reddish
+      case 1:
+        (r = q), (g = v);
+        b = p;
+        break; // yellowish
+      case 2:
+        (r = p), (g = v);
+        b = t;
+        break; // greenish
+      case 3:
+        (r = p), (g = q);
+        b = v;
+        break; // cyanish
+      case 4:
+        (r = t), (g = p);
+        b = v;
+        break; // bluish
+      case 5:
+        (r = v), (g = p);
+        b = q;
+        break; // magenta-ish
+    }
+
+    return ColorValue.from(r, g, b, transparency);
+  }
+
+  /** True if the value of this ColorValue is the same as another ColorValue. */
+  public equals(other: ColorValue): boolean {
+    return this._tbgr === other._tbgr;
+  }
+
+  /** Convert this ColorValue to a string in the form "h,s,l" - i.e 120,50%,50% */
+  public getHslString(): string {
+    const hsl = this.toHSL();
+    return `${hsl.h}, ${hsl.s}%, ${hsl.l}%`;
+  }
+
+  /** Convert this ColorValue to a string in the form "hsl(h,s,l)" - i.e hsl(120,50%,50%). */
+  public toHslString(): string {
+    return `hsl(${this.getHslString()})`;
+  }
+
+  /** Convert this ColorValue to a string in the form "hsla(h,s,l,a)" - i.e hsla(120,50%,50%,0.5). */
+  public toHslaString(): string {
+    const alpha = this.getAlpha() / 255;
+    return `hsla(${this.getHslString()}, ${alpha.toFixed(2)})`;
+  }
+
+  /** Convert this ColorValue to a string in the form "h,s,v" - i.e 120,50%,50% */
+  public getHsvString(): string {
+    const hsv = this.toHSV();
+    return `${hsv.h}, ${hsv.s}%, ${hsv.v}%`;
+  }
+
+  /** Convert this ColorValue to a string in the form "hsv(h,s,v)" - i.e hsv(120,50%,50%). */
+  public toHsvString(): string {
+    return `hsv(${this.getHsvString()})`;
+  }
+
+  /** Convert this ColorValue to a string in the form "hsva(h,s,v,a)" - i.e hsva(120,50%,50%,0.5). */
+  public toHsvaString(): string {
```

Comment:
Hmm... Maybe we could save alpha value from the input and if it is defined, return it with all schemes. That would lessen our code and require just to have `toHSVString`, etc...

Reply:
What if showAlpha is requested but it is set to 1.... in this case tinycolor2 would just return the hsv()/ rgb() functions and not the hsva/rgba - and that may confuse the user who request to see the alpha value.... and I think that is the same functionality you are requesting.